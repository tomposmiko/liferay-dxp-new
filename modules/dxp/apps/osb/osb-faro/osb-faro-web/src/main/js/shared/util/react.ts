import {isEqual} from 'lodash';

export const getDisplayName = WrappedComponent =>
	WrappedComponent.displayName || WrappedComponent.name || 'Component';

type HasChanges = <T>(
	prev: T | object,
	next: T | object,
	...keys: string[]
) => boolean;

/**
 * Compare previous state or props object by provided keys to detect changes.
 */
export const hasChanges: HasChanges = (prev = {}, next = {}, ...keys) => {
	for (const key of keys) {
		if (key in next) {
			const newVal = next[key];

			const prevVal = prev[key];

			if (!isEqual(newVal, prevVal)) {
				return true;
			}
		}
	}

	return false;
};
