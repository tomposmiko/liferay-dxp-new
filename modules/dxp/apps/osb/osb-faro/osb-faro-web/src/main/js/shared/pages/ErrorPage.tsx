import ClayLink from '@clayui/link';
import getCN from 'classnames';
import React from 'react';
import {Routes, toRoute} from 'shared/util/router';

interface IErrorPageProps {
	className?: string;
	href?: string;
	linkLabel?: string;
	message?: string;
	subtitle?: string;
	title?: string;
}

const ErrorPage: React.FC<IErrorPageProps> = ({
	className,
	href = toRoute(Routes.BASE),
	linkLabel = Liferay.Language.get('go-to-home'),
	message = Liferay.Language.get(
		'the-page-you-are-looking-for-does-not-exist'
	),
	subtitle = Liferay.Language.get('page-not-found'),
	title = '404'
}) => (
	<div className={getCN('error-page-root', 'page-container', className)}>
		<h1>{title}</h1>

		<h3>{subtitle}</h3>

		<p>{message}</p>

		<ClayLink
			button
			className='button-root'
			displayType='secondary'
			href={href}
		>
			{linkLabel}
		</ClayLink>
	</div>
);

export default ErrorPage;
