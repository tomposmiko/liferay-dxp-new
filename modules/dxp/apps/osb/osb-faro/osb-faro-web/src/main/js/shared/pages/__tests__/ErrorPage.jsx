import ErrorPage from '../ErrorPage';
import React from 'react';
import {render} from '@testing-library/react';
import {StaticRouter} from 'react-router';

jest.unmock('react-dom');

const DefaultComponent = props => (
	<StaticRouter>
		<ErrorPage {...props} />
	</StaticRouter>
);

describe('ErrorPage', () => {
	it('should render', () => {
		const {container} = render(<DefaultComponent />);

		expect(container).toMatchSnapshot();
	});

	it('should render a custom message', () => {
		const {getByText} = render(<DefaultComponent message='foo bar' />);

		expect(getByText('foo bar')).toBeTruthy();
	});
});
