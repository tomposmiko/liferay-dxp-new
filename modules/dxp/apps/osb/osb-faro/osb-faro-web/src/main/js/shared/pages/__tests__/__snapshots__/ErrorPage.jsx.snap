// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should render 1`] = `
<div>
  <div
    class="error-page-root page-container"
  >
    <h1>
      404
    </h1>
    <h3>
      Page Not Found
    </h3>
    <p>
      The page you are looking for does not exist.
    </p>
    <a
      class="button-root btn btn-secondary"
      href="/"
    >
      Go to Home
    </a>
  </div>
</div>
`;
